src/master-playlist-controller.js

```diff
@@ -388,33 +389,32 @@ export default class MasterPlaylistController extends videojs.EventTarget {
     let media = this.masterPlaylistLoader_.media();
 
     // check that everything is ready to begin buffering
-
     // 1) the active media playlist is available
     if (media &&
-
         // 2) the video is a live stream
         !media.endList &&
 
-        // 3) the player has not played before and is not paused
-        this.tech_.played().length === 0 &&
-        !this.tech_.paused()) {
+        // 3) the player is not paused
+        !this.tech_.paused() &&
 
-      this.load();
+        // 4) the player has not started playing
+        !this.hasPlayed_) {
 
-      // 4) the video element or flash player is in a readyState of
-      // at least HAVE_FUTURE_DATA
-      if (this.tech_.readyState() >= 1) {
```

Comment:
I think I remember that the reason we had the readyState check was because firefox expected a segment to be loaded before a certain action took place. Was it loadedmetadata firing (which I know we handle ourselves now), or was it the ability to do a seek?